### DISCUSSION TITLE
Exam after each lesson

### DISCUSSION THREAD

**Discussion by marioiordanov** [Cyfrin/foundry-full-course-cu#906] [Ideas]
Why not having an exam of 10-15 questions, after each lesson, or 3-4 questions after each mini-lesson. The questions shall be medium-hard IMO.

**Comment by theirrationalone** [comment]
You have challenges as Exams after each lesson. Complexity of each challenge also involves sub lesson knowledge 

;`)

**Comment by marioiordanov** [comment]
Well I mean about the blockchain basics lesson. There is no such exam, but
it shall be because its the foundation

**Reply by theirrationalone** [reply]
@npa528 has created this awesome index for all of us [index of all challenges](https://github.com/Cyfrin/foundry-full-course-f23/discussions/135) . There, You can clearly see there is an exam/challenge for lesson 1: The Basics. However Lesson 0 is just an introduction for us about the course.

**Reply by marioiordanov** [reply]
Thank you for this index. Maybe I don't get what is the challenge for Lesson 1, If its how to write your twitter id into a smart contract, then I think this is not the way to test your knowledge about blockchain theoretic knowledge

**Reply by theirrationalone** [reply]
Try to walk through the challenge code, do a little research about them. You'll get the basics.

**Comment by Niferu** [comment]
Hi @marioiordanov 

None will be checking your progress it is free course you are doing it for yourself, your exam will be coding live projects ;)

Additionally challenges mentioned by @theirrationalone starts easy but later few of them can give you some serious nightmares
